### PR TITLE
Feat docker compose

### PR DESCRIPTION
**Pull request by jotabulacios**
This PR establishes a dockerized environment 
It incorporates Dockerfiles for the following components:
- Opetator 
- Aggretor
- Anvil tesnet

## Type of change

- [x] New 

To test it, run:

```bash
docker-compose build
```

then

```bash
docker-compose up
```

### CODE REVIEW

File: contracts/scripts/anvil/state/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM ubuntu:22.04 as base_system
+
+# Update packages and install required dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y python3 python3-pip git nano curl iputils-ping redis-tools autossh nmap apt-utils supervisor
```
Comment (taturosati) [inline contracts/scripts/anvil/state/Dockerfile]: Why installing so many dependencies?

File: aggregator/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Download Go dependencies
+RUN go mod download
+
+# Copy the rest of the application files
+COPY . .
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y python3 python3-pip git nano curl iputils-ping redis-tools autossh nmap apt-utils supervisor
```
Comment (taturosati) [inline aggregator/Dockerfile]: Why installing so many dependencies?

File: operator/config.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  earnings_receiver_address: 0x7656302c27ec57d6b177a606614257808f63821e
+  delegation_approver_address: "0x0000000000000000000000000000000000000000"
+  staker_opt_out_window_blocks: 0
+  metadata_url: "https://yetanotherco.github.io/operator_metadata/metadata.json"
+el_delegation_manager_address: "0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9"
+eth_rpc_url: http://localhost:8545
+private_key_store_path: "/Users/franco/.eigenlayer/operator_keys/operator.ecdsa.key.json"
+bls_private_key_store_path: "/Users/franco/.eigenlayer/operator_keys/operator.bls.key.json"
+signer_type: local_keystore
+chain_id: 31337
```
Comment (taturosati) [inline operator/config.yaml]: Add missing new line

File: operator/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+COPY . .
+# Build the application binary.
+# You can specify the main package with the directory containing the 'main.go' file.
+RUN CGO_ENABLED=0 GOOS=linux go build -o operator-app ./operator/cmd
+# Start from a new stage to create a smaller image.
+FROM scratch
+# Copy the binary from the builder stage to the current stage.
+COPY --from=builder /app/operator-app /operator-app
+# Set the binary as the entrypoint of the container.
+ENTRYPOINT ["/operator-app"]
```
Comment (taturosati) [inline operator/Dockerfile]: Need to update to pass in config-file flag

Comment (taturosati) [reply operator/Dockerfile]: + Missing new line at end of file

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      - devnet
+    network_mode: host
+
+  operator-3:
+    build:
+      context: ./
+      dockerfile: operator/Dockerfile
+    depends_on:
+      - devnet
+    network_mode: host
```
Comment (JuArce) [inline docker-compose.yml]: missing new line

**Review by JuArce — CHANGES_REQUESTED** [review]

File: contracts/scripts/anvil/state/Dockerfile (hunk 1 above)
Comment (isoprophlex) [reply contracts/scripts/anvil/state/Dockerfile]: I deleted a few a seconds ago, but python, git and curl are necessary to install foundry or eigenlayer.

File: aggregator/Dockerfile (hunk 2 above)
Comment (isoprophlex) [reply aggregator/Dockerfile]: I deleted a few a seconds ago, but python, git and curl are necessary to install foundry or eigenlayer.

**Comment by MauroToscano** [comment]
Stale

**Pull request closed without merging**